validator/src/validator.rs

```diff
@@ -132,7 +134,8 @@ impl<TValidatorNetwork: ValidatorNetwork> Validator<TValidatorNetwork>
 where
     PubsubId<TValidatorNetwork>: std::fmt::Debug + Unpin,
 {
-    const PRODUCER_TIMEOUT: Duration = Duration::from_millis(Policy::BLOCK_PRODUCER_TIMEOUT);
+    const MAXIMUM_PRODUCER_TIMEOUT: u64 = 30_000; // 30 seconds
```

Comment:
30 seconds maximum timeout seems excessive. I think this should be between 12 and 16 seconds, i.e. 3-4 times the minimum timeout to not allow the chain to slow down too much.

Also, typing this as `Duration` instead of `u64` seems more expressive.
```suggestion
    const MAXIMUM_PRODUCER_TIMEOUT: Duration = Duration::from_secs(16);
```

Reply:
It can be done but it always needs to get converted to a `u64` at runtime which kinda defeats the purpose of it being a `const` since it's only being used for clamping. Not a fan of it.

Reply:
We need a `Duration` in the end to pass into `sleep()`. So you can just convert the `u64` timeout into a `Duration` and then use `Duration::clamp` to restrict the range.